### ISSUE TITLE
Support try_from

### ISSUE THREAD

**Issue by TedDriggs** [Voultapher/self_cell#8]
First: Thank you for making this crate - it's really easy to use and has helped me make several of my own APIs simpler.

I have some code that reads a JSON file into a `String` then parses that JSON into a `Thing<'de>` using `#[serde(borrow)]` to avoid thousands/millions of string allocations. I'd like to use `sync_once_self_cell` with it, but I'm not sure how I'd deal with the possibility of the deserialization producing an error.

Ideally, I could do something like this:

```rust
struct Thing<'de>(Vec<Item<'de>>);

impl<'de> TryFrom<&'de String> for Thing<'de> {
    type Error = serde_json::Error;

    fn try_from(s: &'de String) -> Result<Self, Self::Error> {
        serde_json::from_str(s).map(Self)
    }
}

sync_try_once_self_cell(ThingStore, String, Thing<'_>);

fn use_it(s: String) -> Result<ThingStore, serde_json::Error> {
    ThingStore::new(s)
}
```

# Alternatives/Workarounds Considered

* Call `unwrap` in a `From<&'de String> for Thing<'de>` impl. Sadly, the data I'm working with contains errors I can't fix, so this would always crash my program.
* Change `Thing<'de>` to be `Thing<'de>(Result<Vec<Item<'de>>, serde_json::Error>)` - it seems inefficient to have to check if it's `Ok` every time I go to use it.

**Comment by Voultapher** [comment]
The newly released version 0.7 is a ground up redesign and reimplementation. I suggest you switch to it, not only does it address the soundness bugs #4 #5 it also includes new features such as a `try_from` constructor type which covers exactly the problem you ran into. Check out the documentation to learn more about how to use it https://docs.rs/self_cell/0.7.1/self_cell/index.html

**Comment by TedDriggs** [comment]
Perfect, thank you!
